src/Symfony/Component/AssetMapper/ImportMap/ImportMapUpdateChecker.php

```
@@ -0,0 +1,101 @@
+    {
+        $currentVersion = $this->resolver->getPackageVersion($entry->url) ?? $entry->version;
+        $updateInfo = new PackageUpdateInfo($entry->importName, $entry->url, $currentVersion);
+
+        if (ImportMapType::CSS === $entry->type) {
+            return $updateInfo;
+        }
+        $updateInfo->latestVersion = $this->getLatestPackageVersion($entry->importName);
+        $updateInfo->updateType = $this->getUpdateType($currentVersion, $updateInfo->latestVersion);
+
+        return $updateInfo;
+    }
+
+    private function getLatestPackageVersion(string $packageName): string
+    {
+        $response = $this->httpClient->request('GET', sprintf(self::URL_PACKAGE_METADATA, $packageName, ['headers' => ['Accept' => 'application/vnd.npm.install-v1+json']]));
+        $latestVersion = json_decode($response->getContent(), true)['dist-tags']['latest'];
+
+        return $latestVersion;
+    }
+
+    private function getVersionPart(string $version, int $part): ?string
+    {
+        return explode('.', $version)[$part] ?? $version;
+    }
+
+    private function getUpdateType(string $currentVersion, string $latestVersion): string
+    {
+        if (version_compare($currentVersion, $latestVersion, '>')) {
+            return PackageUpdateInfo::UPDATE_TYPE_DOWNGRADE;
+        }
+        if (version_compare($currentVersion, $latestVersion, '==')) {
+            return PackageUpdateInfo::UPDATE_TYPE_UP_TO_DATE;
+        }
+        if ($this->getVersionPart($currentVersion, 0) < $this->getVersionPart($latestVersion, 0)) {
+            return PackageUpdateInfo::UPDATE_TYPE_MAJOR;
+        }
+        if ($this->getVersionPart($currentVersion, 1) < $this->getVersionPart($latestVersion, 1)) {
+            return PackageUpdateInfo::UPDATE_TYPE_MINOR;
+        }
+        if ($this->getVersionPart($currentVersion, 2) < $this->getVersionPart($latestVersion, 2)) {
+            return PackageUpdateInfo::UPDATE_TYPE_PATCH;
+        }
+
+        return 'unknown';
```

Let's throw an exception. If the user gets here, it looks like it would be some bug in our logic, right? I mean: there is no normal way that we can think of right now where this would happen, right?